### ISSUE TITLE
master-failover succeeds, but IP remains assigned to old master

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1113]
Originally reported of Google Code with ID 1057.

```
What software version are you running? Please provide the output of "gnt-cluster --version", "gnt-cluster version", and "hspace --version".

gnt-cluster (ganeti v2.10.1) 2.10.1

hspace (ganeti) version v2.10.1
compiled with ghc 7.4
running on linux x86_64

Software version: 2.10.1
Internode protocol: 2100000
Configuration format: 2100000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.10.1

What distribution are you using? "cat /etc/issue"
Ubuntu 12.04.5 LTS

<b>What steps will reproduce the problem?</b>
1. gnt-cluster master-failover

The master candidate wasn't able to reach the old master node, so node_deactivate_master_ip failed. Consequently the master candidate couldn't acquire the cluster master IP address (with a warning message: "...continuing but activating the master on the current node will probably fail", see attached log), however the command completed as if the failover was successful.

```
Originally added on 2015-03-31 16:44:08 +0000 UTC.

Attachments: 
[master_failover.log](https://storage.googleapis.com/google-code-attachments/ganeti/issue-1057/comment-0/master_failover.log)

**Comment by Ganeti-Issues-Migrator** [comment]
```
Which behavior would be desirable? Note that the actual master failover
did succeed in the sense that the new node now has the role as the
master node---it just didn't get the master IP. Normally, you would
now power down the old master and 'gnt-cluster activate-master-ip'
from the new node.

So, what should gnt-cluster master-failover do in this situation?
I can think of
- writing a more explicit warning, and
- returning a non-zero exit code.
Anything else that would be desirable?

```
Originally added on 2015-04-01 07:24:11 +0000 UTC.

Added to Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-04-01 07:24:33 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
In my personal opinion both actions would be desirable - writing an explicit warning (for humans) and returning a non-zero exit code (for automation). Please note that the warning I mentioned above was found only after digging into the logs when we noticed that something is wrong - there was no warning on the console, the command completed silently.
```
Originally added on 2015-04-01 10:04:42 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-04-01 10:08:01 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 66a5f242d154a3e9ac29a75ca372e578ee37681a
Author: Klaus Aehlig <aehlig@google.com>
Date:   Thu Apr 2 11:46:03 2015 +0200

    Report failure to deactivate old master IP in exit code
    
    If we failed to disable the old master IP, the master failover
    did not fully succeed, hence that should be reported in the
    exit code. Nevertheless, the best is to proceed, as it is
    better to have working cluster, albeit only reachable via
    the primary IP of the new master node (and not the cluster
    master IP), than not have a cluster. Also note, that for
    this reason master will start up even if it cannot set
    the cluster master IP. So, while there, fix the warning
    message as well.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit fc4bfcca713214057826522a904dabf6f9c8b54b
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Apr 1 18:04:59 2015 +0200

    Expose warnings during master-failover
    
    During master failover, there are some situations where problems
    occur but the best thing to do is to carry on. These problems
    are logged using the usual mechanism. However, a user usually
    does not look into the log file unless the command executed
    returns some hints that something might have gone wrong.
    So also return the warnings as an additional return value,
    allowing the CLI to report properly.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

```
Originally added on 2015-04-02 11:07:21 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 2f1d9bd0c86006e890436329966ee4ec32bffaf1
Author: Klaus Aehlig <aehlig@google.com>
Date:   Thu Apr 2 15:26:45 2015 +0200

    After master-failover verify reachability of master IP
    
    ...and warn if it is not. Note that the master activates
    the master IP in an asynchronous task and will continue
    even if that fails.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

```
Originally added on 2015-04-02 14:04:53 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:40:20 +0000 UTC.

Changed State: Released
